### PR TITLE
Extract suite sync from device

### PR DESCRIPTION
**Pull request by peter-sanderson**
In this PR I am doing what shall have been done long ago moving the SuiteSync data into SuiteSync reducer, not polluting the Device.

🔍🖥️ **Suite desktop test results:** [View in Currents](https://app.currents.dev/projects/4ytF0E/runs?branch=extract-suite-sync-from-device&lookback=7)

🔍🖥️ **Suite web test results:** [View in Currents](https://app.currents.dev/projects/Og0NOQ/runs?branch=extract-suite-sync-from-device&lookback=7)

🔍🖥️ **Suite native android test results:** [View in Currents](https://app.currents.dev/projects/iUe1Y4/runs?branch=extract-suite-sync-from-device&lookback=7)

### CODE REVIEW

File: packages/suite/src/actions/suite/__tests__/storageActions.test.ts [hunk 1]
```diff
@@ -107,6 +113,10 @@ const getInitialState = (prevState?: Partial<PartialState>, action?: any) => ({
         prevState ? prevState.suite : undefined,
         action || ({ type: 'foo' } as any),
     ),
+    suiteSync: suiteSyncReducer(
+        prevState ? prevState.suiteSync : undefined,
+        action || ({ type: 'foo' } as any),
+    ),
```
Comment (Lemonexe) [inline packages/suite/src/actions/suite/__tests__/storageActions.test.ts:119]: 💭  side note: we could just use the desktop `initialAppState` instead of calling reducers with {type:'foo'}

File: suite-common/suite-sync/src/suiteSyncSlice.ts [hunk 2]
```diff
@@ -77,6 +82,31 @@ export const suiteSyncSlice = createSlice({
...
+        setSuiteSyncOwner: (
+            state,
+            {
+                payload,
+            }: PayloadAction<{
+                deviceStaticId: StaticSessionId;
+                owner: EncryptedHex<SuiteSyncOwnerSerialized> | null;
+            }>,
+        ) => {
+            if (payload.owner === null) {
```
Comment (Lemonexe) [inline suite-common/suite-sync/src/suiteSyncSlice.ts:88]: wouldn't it be clearer to split `removeSuiteSyncOwner` from `setSuiteSyncOwner`? Setting null to delete it is not very semantic – seems to me as a relict from moving it from a `device` property.

File: suite-common/suite-sync/src/suiteSyncSlice.ts [hunk 3]
```diff
@@ -77,6 +82,31 @@ export const suiteSyncSlice = createSlice({
                 state.suiteSyncErrors[payload.deviceStaticSessionId] = payload.error;
             }
         },
+        setSuiteSyncOwner: (
+            state,
+            {
+                payload,
+            }: PayloadAction<{
```
Comment (Lemonexe) [inline suite-common/suite-sync/src/suiteSyncSlice.ts]: nit: extract type

File: packages/suite/src/middlewares/suite/suiteMiddleware.ts [hunk 4]
```diff
@@ -30,7 +30,6 @@ const isActionDeviceRelated = (action: AnyAction): boolean => {
             deviceActions.forgetDevice,
             // ?
             deviceActions.setDeviceState,
-            deviceActions.setSuiteSyncOwner,
```
Comment (Lemonexe) [inline packages/suite/src/middlewares/suite/suiteMiddleware.ts:33]: all the more reason to extract entitites from `device`, having them point to device with foreign key relationship, rather than bloating the device object.
Managing the storage of device object was cumbersome as hell..

File: suite-common/suite-sync/src/tests/suiteSyncSlice.test.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                device: mockSuiteDevice({
+                    state: { staticSessionId: DEVICE_STATIC_SESSION_ID_123 },
+                }),
+            }),
+        );
+
+        expect(nextState.suiteSyncOwners[DEVICE_STATIC_SESSION_ID_123]).toBeUndefined();
+        expect(nextState.suiteSyncErrors[DEVICE_STATIC_SESSION_ID_123]).toBeUndefined();
+    });
+});
```
Comment (Lemonexe) [inline suite-common/suite-sync/src/tests/suiteSyncSlice.test.ts:32]: good thing to cover this action with test 👍 
💭  I don't think it's valuable to cover basic action that reducer does, because such a test just duplicates what is already defined in reducer, and tests implementation rather than result.
 This, on the other hand, tests a non-trivial interaction with other reducers, one which would be easy to accidentally break.

**Review by Lemonexe — APPROVED** [review]
Codewise LGTM
- Redux changes
- Tests
- Storage loading and persisting LGTM, though I did not test.
- Mobile persistence should work more easily, though I did not test also.

File: packages/suite/src/actions/suite/__tests__/storageActions.test.ts (hunk 1 above)
Comment (peter-sanderson) [reply packages/suite/src/actions/suite/__tests__/storageActions.test.ts:119]: ah, thanks .. now i understand, but for another day

File: suite-common/suite-sync/src/suiteSyncSlice.ts (hunk 2 above)
Comment (peter-sanderson) [reply suite-common/suite-sync/src/suiteSyncSlice.ts:88]: Maybe? I am just moving it now so we have dependencies right

File: packages/suite/src/actions/suite/__tests__/storageActions.test.ts (hunk 1 above)
Comment (Lemonexe) [reply packages/suite/src/actions/suite/__tests__/storageActions.test.ts:119]: yeah I could have written, that this can be done later :slightly_smiling_face:

File: suite-common/suite-sync/src/suiteSyncSlice.ts (hunk 3 above)
Comment (peter-sanderson) [reply suite-common/suite-sync/src/suiteSyncSlice.ts]: will do

File: suite-common/suite-sync/src/tests/suiteSyncSlice.test.ts (hunk 5 above)
Comment (peter-sanderson) [reply suite-common/suite-sync/src/tests/suiteSyncSlice.test.ts:32]: Codex is gigachad :D

File: suite-native/state/src/reducers.ts [hunk 6]
```diff
@@ -316,7 +316,7 @@ export const prepareRootReducers = (deps: PrepareRootReducersDeps) => {
 
     const suiteSyncPersistedReducer = preparePersistReducer({
         reducer: suiteSyncReducer,
-        persistedKeys: ['settings'],
+        persistedKeys: ['settings', 'suiteSyncOwners'],
```
Comment (juriczech) [inline suite-native/state/src/reducers.ts:319]: 👍

**Review by juriczech — APPROVED** [review]
Great improvement. LGTM
